src/plans/D3MAaveTypeBufferPlan.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+interface TokenLike {
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface ATokenLike {
+    function UNDERLYING_ASSET_ADDRESS() external view returns (address);
+    function ATOKEN_REVISION() external view returns (uint256);
+}
+
+/**
+ *  @title D3M Aave Buffer Plan
+ *  @notice Ensure `buffer` amount of DAI is always available to borrow.
+ *  @dev This plan can be used with both V2 and V3 versions of the Aave codebase.
+ */
+contract D3MAaveTypeBufferPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    uint256                      public buffer;
+
+    TokenLike  public immutable dai;
+    ATokenLike public immutable adai;
+    uint256    public immutable adaiRevision;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+
+    constructor(address adai_) {
+        adai = ATokenLike(adai_);
+        dai = TokenLike(adai.UNDERLYING_ASSET_ADDRESS());
+        adaiRevision = adai.ATOKEN_REVISION();
+        
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MAaveTypeBufferPlan/not-authorized");
+        _;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "buffer") {
+            buffer = data;
+        } else revert("D3MAaveTypeBufferPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        if (buffer == 0) return 0; // De-activated
+
+        uint256 liquidityAvailable = dai.balanceOf(address(adai));
+        if (buffer >= liquidityAvailable) {
```

Comment:
I find this naming confusing.  To me the buffer is the additional amount, but the way it is used here is the target total liquidity.  This and other parts of this function could be reworked to say `if (currentAssets < liquidityAvailable + buffer) { // add liquidity`.  I think that would make the code more understandable and provide a better way for governance/risk to set and understand the `buffer` setting

Reply:
The proposed change would be different logic though. `liquidityAvailable` is the amount of DAI in the pool and that is what this Plan attempts to make a fixed value. So say if `buffer` is `50m` then it will try to keep the amount of dai balance in the pool at `50m`.

I can rename the variables if it's confusing, but `liquidityAvailable` is pretty clear to me it's the amount of liquid DAI (IE not being borrowed). `Total market size = Total Debt + Liquidity Available`

LMK if you have better alternatives to names, but the logic should be correct as-is.

Reply:
ah ok, that sounds more like a targetLiq or targetAmt.  As I said, buffer is the cushion amount so imo it creates confusion calling the total a buffer.  To try to keep some of the maker memes, maybe `mark` or `goal`

Reply:
But the total is not `buffer`. In the example above `buffer == 50m`, and it's seeing if the `liquidityAvailable` is above or below `50m` (the buffer). Adding the `currentAssets` in the return is just because the function return expects it in that absolute number format.

Reply:
ok, yes I see what you're saying and how I misinterpreted it. So `buffer` is the target liquidity Maker Governance wants in the your spark system. I think with that understanding i'm fine keeping the name as is, but also feel like it kind of proves the value of daiwanese to avoid my confusion preconceptions... maybe `buff` is a good middle ground, but i'd say unless @gbalabasquer or @rockyfour want to tip the scale for a change, lets just leave it

Reply:
`buff` seems a good name and shouldn't be that problematic to change it.

Reply:
Any issue with just leaving it? `buffer` and `buff` are the same thing to me. :)

Reply:
Yeah I guess if Chris doesn't want to pursue with it, I'd be fine leaving as it.